src/Cli/Microsoft.TemplateEngine.Cli/Commands/SharedOptionsFactory.cs

```diff
@@ -137,7 +137,8 @@ internal static Option<string[]> CreateColumnsOption()
                 TabularOutputSettings.ColumnNames.Author,
                 TabularOutputSettings.ColumnNames.Language,
                 TabularOutputSettings.ColumnNames.Type,
-                TabularOutputSettings.ColumnNames.Tags);
+                TabularOutputSettings.ColumnNames.Tags,
```

Comment:
this option is now reused for `dotnet new list` and `dotnet new search`. `Downloads` are only applicable for `dotnet new search`, therefore please create two separate options now.

Reply:
Do you want these two separate options created in this same `SharedOptionsFactory` or another class? Wasn't sure if having them both here defeats the purpose of a 'shared' class(?)

Reply:
imo, they can now live in respective command implementations

Reply:
I'm a bit stuck on how to handle this. There's certain instances for example [here](https://github.com/dotnet/sdk/blob/7cbb54bdc175e5ff8061d47675e9a36b62b15dc1/src/Cli/Microsoft.TemplateEngine.Cli/Commands/NewCommand.Legacy.cs#L43), where the usage is on the parent command. I tried removing the `ColumnsOption` from this class but I get a failure (_Unrecognized command or argument(s): 'tags'._) when trying to run `dotnet new --columns tags --tag Common --search`

Reply:
This got a bit hairy due to the usage in the `NewCommand.Legacy`. I opted to use `AcceptOnlyFromAmong` column options defined separately for search vs list. Lemme know if this makes sense or how to handle this differently.

Reply:
It's getting a bit hairy with the `NewCommand.Legacy` and the legacy list and search referencing the same `ParentCommand`. I tried to use `AcceptOnlyFromAmong`, but that seems to clear the allowed options, such that when I try `dotnet new console --columns downloads --search`, I get _Argument 'downloads' not regonized. Must be one of: 'author' 'language' 'type' 'tags'_. Any suggestions on how to go about this?